src/libraries/System.Runtime.InteropServices/tests/System.Runtime.InteropServices.UnitTests/System/Runtime/InteropServices/NativeMemoryTests.cs

```diff
@@ -558,5 +558,43 @@ public void ClearWithNullPointerAndZeroByteCountTest()
 
             // This test method just needs to check that no exceptions are thrown
         }
+
+        [Fact]
+        public void CopyNullBlockShouldNoOp()
+        {
+            // This should not throw
+            NativeMemory.Copy(null, null, 0);
+        }
+
+        [Fact]
+        public void CopyEmptyBlockShouldNoOp()
+        {
+            int* source = stackalloc int[1] { 42 };
+            int* destination = stackalloc int[1] { 0 };
+
+            NativeMemory.Copy(source, destination, 0);
+
+            Assert.Equal(0, destination[0]);
+        }
+
+        [Theory]
+        [InlineData(1, 1, 1)]
+        [InlineData(1, 16, 1)]
+        [InlineData(16, 16, 16)]
+        [InlineData(1024, 16, 16)]
```

Comment:
I'd like to see some values that aren't related to powers of 2.

Reply:
Fixed